sandbox/modules/identity/src/internalClusterTest/java/org/opensearch/identity/BasicAuthenticationIT.java

```diff
@@ -0,0 +1,186 @@
+import org.opensearch.client.Response;
+import org.opensearch.client.RestClient;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.plugins.NetworkPlugin;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.test.OpenSearchIntegTestCase.ClusterScope;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.Transport;
+import org.opensearch.transport.TransportInterceptor;
+import org.opensearch.transport.TransportMessageListener;
+import org.opensearch.transport.TransportRequest;
+import org.opensearch.transport.TransportRequestOptions;
+import org.opensearch.transport.TransportResponse;
+import org.opensearch.transport.TransportResponseHandler;
+import org.opensearch.transport.TransportService;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertNoTimeout;
+
+@ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class BasicAuthenticationIT extends HttpSmokeTestCaseWithIdentity {
+
+    public static Map<String, String> interceptedTokens = new HashMap<>();
+    private final static String expectedActionName = "cluster:monitor/health";
+
+    public static class TokenInterceptorPlugin extends Plugin implements NetworkPlugin {
+        public TokenInterceptorPlugin() {}
+
+        @Override
+        public List<TransportInterceptor> getTransportInterceptors(
+            NamedWriteableRegistry namedWriteableRegistry,
+            ThreadContext threadContext
+        ) {
+            return Arrays.asList(new TransportInterceptor() {
+                @Override
+                public AsyncSender interceptSender(AsyncSender sender) {
+                    return new AsyncSender() {
+                        @Override
+                        public <T extends TransportResponse> void sendRequest(
+                            Transport.Connection connection,
+                            String action,
+                            TransportRequest request,
+                            TransportRequestOptions options,
+                            TransportResponseHandler<T> handler
+                        ) {
+
+                            Map<String, String> tcHeaders = threadContext.getHeaders();
+                            if (expectedActionName.equals(action)) {
+                                if (tcHeaders.containsKey(ThreadContextConstants.OPENSEARCH_AUTHENTICATION_TOKEN_HEADER)) {
+                                    interceptedTokens.put(
+                                        request.getParentTask().getNodeId(),
```

Comment:
Might be better to make an object 'InterceptedTokenData' that has all the details, then you can filter by action / whatever afterward.